code/functions/CMakeLists.txt

```diff
@@ -5,9 +5,16 @@ project( functions LANGUAGES CXX )
 
 # Set up the compilation environment.
 include( "${CMAKE_CURRENT_SOURCE_DIR}/../CompilerSettings.cmake" )
+include( "${CMAKE_CURRENT_SOURCE_DIR}/../SolutionTarget.cmake" )
 
 # Create the user's executable.
 add_executable( functions "Structs.h" "Structs.cpp" "functions.cpp" )
+
+# Create the "solution executable".
+add_executable( functions.sol EXCLUDE_FROM_ALL "Structs.h" "Structs.cpp" "solution/functions.sol.cpp" )
+target_include_directories(functions.sol PUBLIC ${CMAKE_SOURCE_DIR})
+add_dependencies( solution functions.sol )
+
 if( "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" )
-   target_compile_definitions( functions PRIVATE _CRT_SECURE_NO_WARNINGS )
+   target_compile_definitions( functions functions.sol PRIVATE _CRT_SECURE_NO_WARNINGS )
```

Comment:
You will need to do this one 2 separate lines/calls. The [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html) function only affects one target at a time.

Reply:
Ah, I should have checked ...